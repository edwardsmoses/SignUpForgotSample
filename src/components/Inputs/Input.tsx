import React from 'react';
import { StyleSheet, TextInput, View } from 'react-native';

import * as Colors from '../../styles/Color';

interface InputProps {
    placeholder: string;
    value: string;
    autoFocus: boolean;
    isPasswordEntry: boolean;
}

const Input: React.FC<InputProps> = ({
    placeholder,
    value,
    autoFocus,
    isPasswordEntry,
}) => {
    return (
        <View>
            <TextInput
                placeholder={placeholder}
                autoCapitalize="none"
                autoFocus={autoFocus || false}
                secureTextEntry={isPasswordEntry || false}
                style={styles.textInputStyle}
                value={value}
            />
        </View>
    );
};

export default Input;

const styles = StyleSheet.create({
    textInputStyle: {
        borderBottomWidth: 1,
        borderWidth: 1,
        borderColor: Colors.Grey,
        marginVertical: 10,
        borderRadius: 5,
    },
});
