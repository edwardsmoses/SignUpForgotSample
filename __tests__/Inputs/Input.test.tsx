import { render, cleanup } from 'react-native-testing-library';
import React from 'react';
import Input from '../../src/components/Inputs/Input';

afterEach(cleanup);

describe('<Input />', () => {
    it('should match snapshot', () => {
        const rendered = render(
            <Input
                placeholder="Password"
                value=""
                autoFocus={true}
                isPasswordEntry={true}
            />,
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });
});
