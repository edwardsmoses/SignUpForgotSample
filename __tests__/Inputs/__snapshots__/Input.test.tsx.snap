// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoFocus={true}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#6d6d6d",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginVertical": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
